tests/test_ManifestReader/test_manifestfilereader_load_manifest.py

```diff
@@ -27,31 +27,53 @@
 import pytest
 
 from smartdashboard.utils.errors import ManifestError, VersionIncompatibilityError
-from smartdashboard.utils.ManifestReader import Manifest, load_manifest
+from smartdashboard.utils.ManifestReader import (
+    Manifest,
+    ManifestFileReader,
+    load_manifest,
+)
 
 
 @pytest.mark.parametrize(
-    "json_file, result_type",
+    "json_file, manifest_type, reader_type",
     [
-        pytest.param("tests/utils/manifest_files/manifesttest.json", Manifest),
-        pytest.param("tests/utils/manifest_files/0.0.2_manifest.json", Manifest),
-        pytest.param("tests/utils/manifest_files/no_apps_manifest.json", Manifest),
-        pytest.param("file_doesn't_exist.json", ManifestError),
-        pytest.param("tests/utils/manifest_files/JSONDecodererror.json", ManifestError),
+        pytest.param(
+            "tests/utils/manifest_files/manifesttest.json", Manifest, ManifestFileReader
+        ),
+        pytest.param(
+            "tests/utils/manifest_files/0.0.2_manifest.json",
+            Manifest,
+            ManifestFileReader,
+        ),
+        pytest.param(
+            "tests/utils/manifest_files/no_apps_manifest.json",
+            Manifest,
+            ManifestFileReader,
+        ),
+        pytest.param("file_doesn't_exist.json", ManifestError, None),
+        pytest.param(
+            "tests/utils/manifest_files/JSONDecodererror.json", ManifestError, None
+        ),
         pytest.param(
             "tests/utils/manifest_files/invalid_version.json",
             VersionIncompatibilityError,
+            None,
         ),
     ],
 )
-def test_load_manifest(json_file, result_type):
+def test_load_manifest_and_has_changed(json_file, manifest_type, reader_type):
     try:
-        manifest = load_manifest(json_file)
+        manifest, manifest_reader = load_manifest(json_file)
     except ManifestError:
-        assert result_type == ManifestError
+        assert manifest_type == ManifestError
         return
     except VersionIncompatibilityError:
-        assert result_type == VersionIncompatibilityError
+        assert manifest_type == VersionIncompatibilityError
         return
```

Comment:
Not wrong, but if you want guarantee that errors you expect to be thrown are thrown, consider using a `pytest.rasies` context

Reply:
^^ If you do go down that road, I might recommend breaking this test up into two distinct tests: one that checks that expected errors are raise, and one that checks that the `has_changed` attr is working.

Reply:
You got it!